tests/unit/storage_v2_indices.cpp

```diff
@@ -1673,3 +1677,405 @@ TYPED_TEST(IndexTest, EdgeTypeIndexRepeatingEdgeTypesBetweenSameVertices) {
+              IsEmpty());
+
+  EXPECT_THAT(this->GetIds(acc_after->Edges(this->edge_type_id1, this->edge_prop_id1, View::NEW), View::NEW),
+              IsEmpty());
+
+  EXPECT_THAT(this->GetIds(acc_after_commit->Edges(this->edge_type_id1, this->edge_prop_id1, View::NEW), View::NEW),
+              UnorderedElementsAre(0, 1, 2, 3, 4));
+}
+
+// NOLINTNEXTLINE(hicpp-special-member-functions)
+TYPED_TEST(IndexTest, EdgeTypePropertyIndexCountEstimate) {
+  if constexpr (!(std::is_same_v<TypeParam, memgraph::storage::InMemoryStorage>)) {
+    return;
+  }
+  {
+    auto unique_acc = this->storage->UniqueAccess(ReplicationRole::MAIN);
+    EXPECT_FALSE(unique_acc->CreateIndex(this->edge_type_id1, this->edge_prop_id1).HasError());
+    ASSERT_NO_ERROR(unique_acc->Commit());
+  }
+  {
+    auto unique_acc = this->storage->UniqueAccess(ReplicationRole::MAIN);
+    EXPECT_FALSE(unique_acc->CreateIndex(this->edge_type_id2, this->edge_prop_id2).HasError());
+    ASSERT_NO_ERROR(unique_acc->Commit());
+  }
+
+  auto acc = this->storage->Access(ReplicationRole::MAIN);
+  for (int i = 0; i < 20; ++i) {
+    auto vertex_from = this->CreateVertexWithoutProperties(acc.get());
+    auto vertex_to = this->CreateVertexWithoutProperties(acc.get());
+    if (i % 3) {
+      auto edge_acc = this->CreateEdge(&vertex_from, &vertex_to, this->edge_type_id1, acc.get());
+      edge_acc.SetProperty(this->edge_prop_id1, memgraph::storage::PropertyValue(i));
+    } else {
+      auto edge_acc = this->CreateEdge(&vertex_from, &vertex_to, this->edge_type_id2, acc.get());
+      edge_acc.SetProperty(this->edge_prop_id2, memgraph::storage::PropertyValue(i));
+    }
+  }
+
+  EXPECT_EQ(acc->ApproximateEdgeCount(this->edge_type_id1, this->edge_prop_id1), 13);
+  EXPECT_EQ(acc->ApproximateEdgeCount(this->edge_type_id2, this->edge_prop_id2), 7);
+}
+
+// NOLINTNEXTLINE(hicpp-special-member-functions)
+TYPED_TEST(IndexTest, EdgeTypePropertyIndexRepeatingEdgeTypesBetweenSameVertices) {
+  if constexpr (!(std::is_same_v<TypeParam, memgraph::storage::InMemoryStorage>)) {
+    return;
+  }
+  {
+    auto unique_acc = this->storage->UniqueAccess(ReplicationRole::MAIN);
+    EXPECT_FALSE(unique_acc->CreateIndex(this->edge_type_id1, this->edge_prop_id1).HasError());
+    ASSERT_NO_ERROR(unique_acc->Commit());
+  }
+
+  auto acc = this->storage->Access(ReplicationRole::MAIN);
+  auto vertex_from = this->CreateVertexWithoutProperties(acc.get());
+  auto vertex_to = this->CreateVertexWithoutProperties(acc.get());
+
+  for (int i = 0; i < 5; ++i) {
+    auto edge_acc = this->CreateEdge(&vertex_from, &vertex_to, this->edge_type_id1, acc.get());
+    edge_acc.SetProperty(this->edge_prop_id1, memgraph::storage::PropertyValue(i));
+  }
+
+  EXPECT_EQ(acc->ApproximateEdgeCount(this->edge_type_id1, this->edge_prop_id1), 5);
+
+  EXPECT_THAT(this->GetIds(acc->Edges(this->edge_type_id1, this->edge_prop_id1, View::NEW), View::NEW),
+              UnorderedElementsAre(0, 1, 2, 3, 4));
+}
```

Comment:
I think we also need to add changing property value on indexed edge type + property to test UpdateOnModifiedEdge